app/front/scripts/controllers/main.js

```diff
@@ -69,7 +73,24 @@ angular.module('Application')
       function initRegular() {
         $q(LoginService.tryGetToken())
           .then(function(token) {
-            return $q(osViewerService.loadDataPackages(token));
+            var packageId = null;
+            var userId = token ? LoginService.getUserId() : null;
+            if (!token) {
```

Comment:
From a UX point of view, I think the dropdown should always only show the user's owned packages for logged in users and be disabled otherwise.
The viewer should point to the package pointed to in the URL. I'm okay with redirecting to explorer if there's no package id in the URL (i.e. no auto-select of package from list).
I'm also okay with always doing separate queries to the search API if that simplifies code.